app/src/main/java/org/mozilla/focus/shortcut/IconGenerator.kt

```diff
@@ -55,6 +57,25 @@ class IconGenerator {
             return drawCharacterOnBitmap(context, character, shape)
         }
 
+        @JvmStatic
+        fun generateSearchEngineIcon(context: Context): Bitmap {
+            val options = BitmapFactory.Options()
+            options.inMutable = true
+            val shape = BitmapFactory.decodeResource(context.resources, R.drawable.ic_search_engine_shape, options)
+            return drawVectorOnBitmap(context, R.drawable.ic_search, shape, 0.15)
+        }
+
+        private fun drawVectorOnBitmap(context: Context, vectorId: Int, bitmap: Bitmap, frame: Double): Bitmap {
+            val canvas = Canvas(bitmap)
+            // Select the area to draw with a frame
+            val rect = Rect((frame * canvas.width).toInt(), (frame * canvas.height).toInt(), ((1 - frame) * canvas.width).toInt(), ((1 - frame) * canvas.height).toInt())
```

Comment:
\[detekt\] INFO: \[style/MaxLineLength\] Line detected that is longer than the defined maximum line length in the code style\.